pkg/install/util.go

```diff
@@ -37,6 +41,49 @@ func GetPort(cn *corev1.Container, portName string) (*corev1.ContainerPort, erro
 	return nil, fmt.Errorf("unable to locate port %q in container %q", portName, cn.Name)
 }
 
+// GetAppProto determines the application protocol of the given ServicePort. The given AppProtocolStrategy
+// used if the port's appProtocol attribute is unset.
+func GetAppProto(ctx context.Context, aps client.AppProtocolStrategy, p *corev1.ServicePort) string {
+	if p.AppProtocol != nil {
+		appProto := *p.AppProtocol
+		if appProto != "" {
+			dlog.Debugf(ctx, "Using application protocol %q from service appProtocol field", appProto)
+			return appProto
+		}
+	}
```

Comment:
should this function account for a possible `--http-protocol` flag? I'm specifically worried about `--http-protocol=tcp` when the `AppProtocol` is set to, say, `http`.

Reply:
I don't think our agent should override the protocol used when talking to the container app in the cluster it is explicitly stated in the `ports.appProtocol` field.

The `--http-protocol` flag only affects the protocol used during an intercept when forwarding to the interceptor (the process on the local workstation). It has no effect on the communication between the agent and the container app.

Reply:
ah okay got it, makes sense